axonserver/src/test/java/io/axoniq/axonserver/grpc/SubscriptionQueryITCase.java

```diff
@@ -0,0 +1,87 @@
+import org.axonframework.serialization.xml.XStreamSerializer;
+import org.junit.*;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.Reference;
+import java.lang.ref.ReferenceQueue;
+
+import static org.junit.Assert.*;
+
+/**
+ * Integration test for Subscription Queries
+ *
+ * @author Sara Pellegrini
+ * @since 4.2
+ */
+
+public class SubscriptionQueryITCase {
+
+    private final AxonServerConfiguration configuration = AxonServerConfiguration.builder().build();
+
+    private final AxonServerConnectionManager connectionManager = new AxonServerConnectionManager(configuration);
+
+    private final SimpleQueryBus queryBus = SimpleQueryBus.builder().build();
+
+    private final QueryBus axonServerQueryBus = new AxonServerQueryBus(connectionManager,
+                                                                       configuration,
+                                                                       queryBus.queryUpdateEmitter(),
+                                                                       queryBus,
+                                                                       XStreamSerializer.defaultSerializer(),
+                                                                       XStreamSerializer.defaultSerializer(),
+                                                                       QueryPriorityCalculator
+                                                                               .defaultQueryPriorityCalculator());
+
+    @Test
+    public void testMemoryLeak() throws InterruptedException {
+        Registration echo = axonServerQueryBus.subscribe("test", String.class, Message::getPayload);
+        ReferenceQueue<Object> referenceQueue = new ReferenceQueue<>();
+        Reference<Object> ref = new PhantomReference<>(invokeQuery(), referenceQueue);
+        queryBus.queryUpdateEmitter().emit(String.class, c -> true, "update");
+        assertFalse(checkObjectGarbageCollected(ref, referenceQueue));
+        queryBus.queryUpdateEmitter().complete(String.class, c -> true);
+        System.gc();
+        Thread.sleep(1000);
+        System.gc();
+        assertTrue(checkObjectGarbageCollected(ref, referenceQueue));
+        echo.cancel();
+        connectionManager.shutdown();
+    }
+
+    private SubscriptionQueryResult invokeQuery() {
+        GenericSubscriptionQueryMessage<String, String, String> query =
+                new GenericSubscriptionQueryMessage<>("hi",
+                                                      "test",
+                                                      ResponseTypes.instanceOf(String.class),
+                                                      ResponseTypes.instanceOf(String.class));
+        SubscriptionQueryResult<QueryResponseMessage<String>, SubscriptionQueryUpdateMessage<String>> result = queryBus
+                .subscriptionQuery(query);
+
+        result.updates()
+              .subscribe(
+                      System.out::println,
```

Comment:
I'd keep the initial result and incremental updates in memory in order to assert successful invocation of subscription query.